lib/Event/DataConnectorSourceRequestEvent.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/*
+ * Copyright (C) 2024 Xibo Signage Ltd
+ *
+ * Xibo - Digital Signage - https://xibosignage.com
+ *
+ * This file is part of Xibo.
+ *
+ * Xibo is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * Xibo is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Xibo.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace Xibo\Event;
+
+use InvalidArgumentException;
+use Xibo\Connector\DataConnectorSourceProviderInterface;
+
+/**
+ * Event triggered to retrieve a list of data connector sources.
+ *
+ * This event collects metadata (names and IDs) of connectors that provides data connector.
+ */
+class DataConnectorSourceRequestEvent extends Event implements DataConnectorSourceProviderInterface
+{
+    public static $NAME = 'data.connector.source.request';
+
+    /**
+     * @var array
+     */
+    private $dataConnectorSources = [
+        ['id' => 'user_defined', 'name' => 'User Defined JavaScript']
```

Comment:
Can we translate the name?

Reply:
Noted. Sorry, the translation slipped my mind. 😅